### PR TITLE
174789366: fixing scores endpoint so scores aren't returned, only ses…

### PR DESCRIPTION
**Pull request by biotom**
…sion ID 

Do we need an API test for this?

### CODE REVIEW

**Comment by rodriguesk** [comment]
@biotom The API response should have:
#### Response
-- 201 response code indicating a recored has been created for the scoring
-- Session id (so that we can retrieve personal values and feed for this session)

Is it using 201?

**Comment by biotom** [comment]
> @biotom The API response should have:
> #### Response
> 
> -- 201 response code indicating a recored has been created for the scoring
> -- Session id (so that we can retrieve personal values and feed for this session)
> 
> Is it using 201?

Thanks- it is now

**Comment by rodriguesk** [comment]
@biotom 
Good! But now the body submission is still not like how the story asks:
{
  "quesionResponses": {
    "SetOne": [
      { "questionId": 1, "answerId": 1 },
      { "questionId": 2, "answerId": 2 },
      { "questionId": 3, "answerId": 4 },
      { "questionId": 4, "answerId": 5 },
      { "questionId": 5, "answerId": 3 },
      { "questionId": 6, "answerId": 4 },
      { "questionId": 7, "answerId": 2 },
      { "questionId": 8, "answerId": 3 },
      { "questionId": 9, "answerId": 4 },
      { "questionId": 10, "answerId": 3 }
    ]
  }
}

**Comment by NickCallaghan** [comment]
I also believe when I checked earlier that it wasn’t accepting the correct Json body. Can you test in postman that if give it the body attached in the story you get the correct response back? Nick

**Comment by biotom** [comment]
No, sorry, I didn't touch the JSON body part. @seanmajorpayne have you done any work on that?

**Comment by rodriguesk** [comment]
The body JSON is described https://www.pivotaltracker.com/story/show/174789366/comments/218277875
@biotom

**Comment by biotom** [comment]
is there a typo in that json- "quesionResponses"? Should it be question?

**Comment by rodriguesk** [comment]
@biotom yes, that's a typo. Good catch!

**Comment by biotom** [comment]
@rodriguesk sure- I've gone with "questions". Just be sure to update it when things are sent from the frontend, or it won't work. 

I think this should do what we want. I've attached a new screenshot in pivotaltracker. I've checked, and it should also cover the error handling specified in the comments. I have to go to bed, but I can look again tomorrow if there's anything wrong.

**Comment by NickCallaghan** [comment]
> …sion ID
> 
> Do we need an API test for this?

Good work @biotom! So I've got a response back from the api with the sessions id! 🙌

There is still one issue is that the integration tests are conflicting with the front end  as are both trying to use port 5000, and preventing the docker compose from starting. I commented out all the integration tests stuff docker compose worked. 

Not sure if there's any implications of changing the port for the tests.... @mncharlton

**Comment by rodriguesk** [comment]
I approve of the functioning of the code. Regarding the integration and tests, I'll leave that to @mncharlton to give final approval for merging. Things may need to be removed or fixed.

**Comment by NickCallaghan** [comment]
> I approve of the functioning of the code. Regarding the integration and tests, I'll leave that to @mncharlton to give final approval for merging. Things may need to be removed or fixed.

In the short term the front end can develop against this branch so that unblocks us for now. Ideally we want api tests to flag up if things change as that would break any clients using the api. 

Thanks @biotom @rodriguesk We can begin to work on the score submission now!!!! I'm excited.

File: knowledge_graph/routes.py [hunk 1]
```diff
@@ -119,45 +119,59 @@ def receive_user_scores() -> Tuple[Response, int]:
...
 
-    response = Response(dumps(value_scores))
-    return response, 200
+    response = {"session-id": session_id}
+
+    response = Response(dumps(response))
+    return response, 201
 
 
 @app.route("/personal_values", methods=["GET"])
```
Comment (rodriguesk) [inline knowledge_graph/routes.py:177]: Probably need to change the session-id for the "/personal_values" endpoint to not be int() but str()

**Review by NickCallaghan — COMMENTED** [review]
Works for us functionally. Still need to resolve the issue with the ports

**Review by mncharlton — APPROVED** [review]

**Review by mncharlton — COMMENTED** [review]
As others have said - the port conflict needs resolving, that's fine to do imo. We can add an api test afterwards, no need to block frontend work continuing.

**Comment by biotom** [comment]
Ok, I've reformatted the code. I also removed the integration test dockerfile and section of the docker-compose- I added it, and in retrospect I'm not sure it does much for us since the test can be run with pytest. I might be wrong about that!

**Comment by rodriguesk** [comment]
Looks good. Please squash and merge @biotom . Thanks!

**Comment by rodriguesk** [comment]
Feel free to comment with that screenshot you took earlier too.

**Pull request merged**